src/java/org/apache/cassandra/index/sai/IndexContext.java

```diff
@@ -983,4 +977,9 @@ public IndexFeatureSet indexFeatureSet()
         getView().getIndexes().stream().map(SSTableIndex::indexFeatureSet).forEach(set -> accumulator.accumulate(set));
         return accumulator.complete();
     }
+
+    public Version getMinimumVersion()
+    {
+        return getView().getIndexes().stream().map(SSTableIndex::getVersion).min(Version::compareTo).orElse(Version.latest());
```

Comment:
I'm pretty sure we need to explicitly compare the stream result to `latest()`. Otherwise, if we rollback to an earlier version, all current indexes might have one version, then a new index could have an older version.

Reply:
(this was very broken, fixed by passing the exact version of the index instead of using this approach)